### PR TITLE
thunderbird, thunderbird-bin: 45.8.0 -> 52.0

### PR DESCRIPTION
**Pull request by taku0**
###### Motivation for this change

Various enhancements and changes.

“IMPORTANT: The way images are included in a compose window has changed. Images are now included as data URIs and not as references to parts of other messages or operating system files. This allows better interoperability with office packages such as MS Office or LibreOffice. Images linked from locations on the internet will no longer be downloaded and attached to the message automatically. This can be changed for each image individually via the Image Properties dialog or globally by setting the preference mail.compose.attach_http_images.”

https://www.mozilla.org/en-US/thunderbird/52.0/releasenotes/

Related commits: 90f5be31

Note that the calendar add-on (Lightning) does not seem to be compatible with 52 yet.

https://addons.mozilla.org/en-US/thunderbird/addon/lightning/

###### Things done

- [X] Tested using sandboxing
  ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS,
    or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file)
    on non-NixOS)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix [hunk 1]
```diff
@@ -146,6 +151,11 @@ stdenv.mkDerivation {
       GenericName=Mail Reader
       Categories=Application;Network;
       EOF
+
+      wrapProgram "$out/bin/thunderbird" \
+        --argv0 "$out/bin/.thunderbird-wrapped" \
+        --prefix XDG_DATA_DIRS : "$GSETTINGS_SCHEMAS_PATH:" \
+        --suffix XDG_DATA_DIRS : "$XDG_ICON_DIRS"
```
Comment (Mic92) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: I suppose `wrapGAppsHook` can be also used here, right @abbradar?

File: pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix [hunk 2]
```diff
@@ -116,6 +119,8 @@ stdenv.mkDerivation {
       stdenv.cc.cc
     ];
 
+  buildInputs = [ makeWrapper gtk3 defaultIconTheme ];
```
Comment (rasendubi) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: `makeWrapper` should go to `nativeBuildInputs`.

**Comment by rasendubi** [comment]
Passed nox-review locally on NixOS with sandboxing enabled.

File: pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix (hunk 2 above)
Comment (taku0) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: Fixed. I will also fix `firefox-bin` after this PR is merged.

File: pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix (hunk 1 above)
Comment (abbradar) [inline pkgs/applications/networking/mailreaders/thunderbird-bin/default.nix]: Correct!

**Comment by abbradar** [comment]
I'm trying to solve a crash in Thunderbird so I can merge this along with my fixes (most likely we need to disable system `cairo`). I'll also see if we can enable optional GTK3 support.

**Comment by aristidb** [comment]
@abbradar Is the crash in Thunderbird caused by this change?

**Comment by abbradar** [comment]
@aristidb Nope, it's a separate problem, unrelated to this -- I'm just testing things in bulk. I still haven't solved it (plan to finish today or tomorrow) so if anybody wants he/she can merge this earlier.

**Comment by abbradar** [comment]
I've merged it as 55ff94b396b735b145c807fd6ba1191ac3ac1fc1 and 77f0ad5a37c4cc48e9869cd2652bb6d511c3874e. Along with b023370f3757a76acb244e647f29e388e076cab0 this finally fixes crashes that I've experienced for many months now. Thank you!

**Pull request closed without merging**